libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_integration.c

```diff
@@ -45,19 +43,82 @@ extern uint8_t _ucBLEServerIf;
 extern uint16_t _usBLEConnId;
 extern BTBdaddr_t _xAddressConnectedDevice;
 
+extern BTService_t _xSrvcA;
 extern BTService_t _xSrvcB;
 extern uint16_t usHandlesBufferB[ bletestATTR_SRVCB_NUMBER ];
-
+extern response_t ucRespBuffer[ bletestATTR_SRVCB_NUMBER ];
 
 TEST_GROUP( Full_BLE_Integration_Test );
 
 TEST_SETUP( Full_BLE_Integration_Test )
 {
 }
 
+TEST_TEAR_DOWN( Full_BLE_Integration_Test )
+{
+}
+
 /*-----------------------------------------------------------*/
 
-TEST_TEAR_DOWN( Full_BLE_Integration_Test )
+TEST_GROUP( Full_BLE_Integration_Test_common_GATT );
+
+TEST_SETUP( Full_BLE_Integration_Test_common_GATT )
+{
+    GAP_common_setup();
+}
+
+TEST_TEAR_DOWN( Full_BLE_Integration_Test_common_GATT )
+{
+    GAP_common_teardown();
+}
+
+TEST_GROUP_RUNNER( Full_BLE_Integration_Test_common_GATT )
+{
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_GROUP( Full_BLE_Integration_Test_Advertisement );
+
+TEST_SETUP( Full_BLE_Integration_Test_Advertisement )
+{
+    GATT_setup();
+}
+
+TEST_TEAR_DOWN( Full_BLE_Integration_Test_Advertisement )
+{
+    GATT_teardown();
+}
+
+TEST_GROUP_RUNNER( Full_BLE_Integration_Test_Advertisement )
+{
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_GROUP( Full_BLE_Integration_Test_Connection );
+
+TEST_SETUP( Full_BLE_Integration_Test_Connection )
+{
+    /* Initialize */
+    GATT_setup();
+    prvCreateAndStartServiceB();
+
+    /* Advertise and Connect */
+    prvSetAdvProperty();
+    prvSetAdvData( eBTuuidType128 );
+    prvStartAdvertisement();
+    prvWaitConnection( true );
```

Comment:
Functions are marked prv(private) but not in the same file. Move the helper functions to a common file if used by multiple tests.

Reply:
Done